tee-worker/omni-executor/native-task-handler/src/lib.rs

```diff
@@ -791,26 +775,204 @@ async fn handle_native_task<
+					&access_token,
+					chain_id,
+					wallet_index,
+					&recipient_address,
+					&token_ca,
+					&amount,
+					language.clone(),
+				)
+				.await
+			{
+				Ok(res) => res,
+				Err(e) => {
+					send_error(
+						format!("Failed to create_transfer_unsigned_tx: {:?}", e),
+						response_sender,
+						NativeTaskError::PumpxApiError(
+							PumpxApiError::CreateTransferUnsignedTxFailed,
+						),
+					);
+					return;
+				},
+			};
+
+			let transfer_id = create_transfer_res.data.transfer_id;
+			let tx_data = create_transfer_res.data.tx_data;
+			let tx_data = match tx_data {
+				Some(data) => data,
+				None => {
+					send_error(
+						"No tx_data in create_transfer_unsigned_tx response".to_string(),
+						response_sender,
+						NativeTaskError::PumpxApiError(
+							PumpxApiError::CreateTransferUnsignedTxFailed,
+						),
+					);
+					return;
+				},
+			};
+
+			let Some(chain_type) = ChainType::from_pumpx_chain_id(chain_id) else {
+				send_error(
+					format!("Failed to map pumpx chain_id {}", chain_id),
+					response_sender,
+					NativeTaskError::InternalError,
+				);
+				return;
+			};
+
+			// 4. Use the pumpx_signer_client to sign all tx_data entries at once
+			let mut messages_to_sign = Vec::new();
+			for tx in tx_data {
+				let tx_cleaned = tx.strip_prefix("0x").unwrap_or(&tx);
+				let tx_bytes = match hex::decode(tx_cleaned) {
+					Ok(bytes) => bytes,
+					Err(e) => {
+						send_error(
+							format!("Failed to decode tx_data (hex): {:?}", e),
+							response_sender,
+							NativeTaskError::InternalError,
+						);
+						return;
+					},
+				};
+				messages_to_sign.push(tx_bytes);
+			}
+
+			let signatures = match ctx
+				.pumpx_signer_client
+				.request_signatures(
+					chain_type,
+					wallet_index,
+					sender.to_omni_account().into(),
+					messages_to_sign,
+				)
+				.await
+			{
+				Ok(sigs) => sigs,
+				Err(e) => {
+					send_error(
+						format!("Failed to sign transfer tx: {:?}", e),
+						response_sender,
+						NativeTaskError::PumpxSignerError(PumpxSignerError::RequestSignatureFailed),
+					);
+					return;
+				},
+			};
+
+			let signed_tx_data: Vec<String> = signatures.into_iter().map(hex::encode).collect();
+
+			// 5. Send the signed tx to the Pumpx backend
+			match ctx
+				.pumpx_api
+				.send_transfer_tx(&access_token, transfer_id, chain_id, signed_tx_data, language)
+				.await
+			{
+				Ok(res) => {
+					send_ok(response_sender, NativeTaskOk::PumpxTransferWithdraw(res));
+					return;
+				},
+				Err(e) => {
+					send_error(
+						format!("Failed to send_transfer_tx: {:?}", e),
+						response_sender,
+						NativeTaskError::PumpxApiError(PumpxApiError::SendTransferTxFailed),
+					);
+					return;
+				},
+			};
+		},
 	};
-	let report = match rpc_client.submit_and_watch_tx_until(&tx, XtStatus::Finalized).await {
-		Ok(report) => report,
+
+	match rpc_client.submit_and_watch_tx_until(&tx, XtStatus::Finalized).await {
```

Comment:
Why do we need this?
We don't have to report anything to heima, and `tx` isn't a substrate tx I assume?

Reply:
I think it is a substrate tx.
This part is just a refactor.